src/fal/packages/environments/conda.py

```diff
@@ -0,0 +1,125 @@
+
+    @property
+    def requirements(self):
+        # TODO: remove this once cross-Python-version serialization is
+        # working.
+        for raw_requirement in self.conda_requirements:
+            requirement_name, *_ = raw_requirement.partition("=")
+            if "python" == requirement_name:
+                raise RuntimeError(
+                    "Conda environments cannot customize their Python version."
+                )
+
+        return [*self.conda_requirements, f"python={self.python_version}"]
+
+    @property
+    def key(self) -> str:
+        return hashlib.sha256(" ".join(self.requirements).encode()).hexdigest()
+
+    def _get_or_create(self) -> Path:
+        env_path = _BASE_CONDA_DIR / self.key
+        if env_path.exists():
+            return env_path
+
+        with rmdir_on_fail(env_path):
+            self._run_conda(
+                "create",
+                "--yes",
+                # The environment will be created under $BASE_CACHE_DIR/conda
+                # so that in the future we can reuse it.
+                "--prefix",
+                env_path,
+                *self.requirements,
+            )
+
+            # See in conda docs:
+            # manage-environments#using-pip-in-an-environment
+            self._run_conda(
+                "run",
+                "--no-capture-output",
+                "--prefix",
+                env_path,
+                "pip",
+                "install",
+                *get_default_pip_dependencies(),
+            )
+        return env_path
+
+    def _run_conda(self, *args, **kwargs) -> None:
+        log_env(self, "Installing conda environment")
+        conda_executable = get_conda_executable()
+        subprocess.check_call([conda_executable, *args], **kwargs, text=True)
+
+    def open_connection(self, conn_info: Path) -> CondaConnection:
+        conda_path = conn_info
+        return CondaConnection(self, conda_path)
+
+
+@dataclass
+class CondaConnection(IsolatedProcessConnection[CondaEnvironment]):
+    environment_path: Path
+
+    def start_process(
+        self,
+        service: bridge.Listener,
+        *args,
+        **kwargs,
+    ) -> ContextManager[subprocess.Popen]:
+        python_executable = get_executable_path(self.environment_path, "python")
+        return subprocess.Popen(
+            [
+                python_executable,
+                isolated_runner.__file__,
+                bridge.encode_service_address(service.address),
+            ],
+        )
+
+
+@cache_static
+def get_conda_executable() -> Path:
+    for path in [_FAL_CONDA_HOME, None]:
+        conda_path = shutil.which(_CONDA_COMMAND, path=path)
+        if conda_path is not None:
+            return conda_path
+    else:
+        # TODO: we should probably point to the instructions on how you
+        # can install conda here.
+        raise RuntimeError(
+            "Could not find conda executable. Please install conda or set FAL_CONDA_HOME."
```

Comment:
Currently it might, but we don't explicitly support mamba (it is on the todo list though). It might require a bit of more extra work before we have complete support but yes, in theory you can put any executable under `FAL_CONDA_HOME`. The primary use case is for people who don't have a global conda installation (like me) since that tends to shadow everything else.